### PR TITLE
Change emscripten default build dir to "emscriptenBuild"

### PR DESCRIPTION
**Pull request by Marenz**
Avoids complications with local build directory when building.

### CODE REVIEW

**Comment by chriseth** [comment]
ci fails

**Comment by chriseth** [comment]
Also, I would prefer something like `build_emscripten` as a directory name, but that might just be taste.

**Comment by Marenz** [comment]
Fixed & renamed

**Comment by Marenz** [comment]
I am not sure whether I am responsible for that CI error:
```
[ 67%] Building CXX object test/CMakeFiles/soltest.dir/RPCSession.cpp.o
/root/project/test/tools/yulrun.cpp:66:37: error: no member named 'EVMVersion' in namespace 'dev::solidity'
        AssemblyStack stack(dev::solidity::EVMVersion(), AssemblyStack::Language::StrictAssembly);
                            ~~~~~~~~~~~~~~~^
1 error generated.
```

**Comment by axic** [comment]
Rebased.

File: scripts/bytecodecompare/storebytecode.sh [hunk 1]
```diff
@@ -29,6 +29,12 @@ set -e
 
 REPO_ROOT="$(dirname "$0")"/../..
 
+if test -z "$1"; then
+	BUILD_DIR="emscripten_build"
```
Comment (axic) [inline scripts/bytecodecompare/storebytecode.sh]: I would argue this file shouldn't default to `emscripten_build` as this file is not related to emscripten.

Comment (Marenz) [reply scripts/bytecodecompare/storebytecode.sh]: Fair point. I misread the .travis.yml file where the call is.

Comment (Marenz) [reply scripts/bytecodecompare/storebytecode.sh]: Probably also the reason it fails now

**Comment by Marenz** [comment]
updated

**Comment by chriseth** [comment]
What about `test_emscripten.sh`?

**Comment by Marenz** [comment]
I think this is now ready.

**Review by chriseth — APPROVED** [review]
